Modix.Common/Extensions/System/Linq/ModixCommonEnumerableExtensions.cs

```diff
@@ -0,0 +1,23 @@
+﻿using System.Collections.Generic;
+
+namespace System.Linq
+{
+    public static class ModixCommonEnumerableExtensions
+    {
+        public static double? AverageOrNull(
+            this IEnumerable<long> values)
```

Comment:
Why not just use .Average() here?

Reply:
Because `.Average()` throws an exception for an empty dataset. Which can happen if every endpoint fails, since we filter out null results.